### ISSUE TITLE
False positive: S1854 - Remove this useless assignment...

### ISSUE THREAD

**Issue by remcoros** [SonarSource/sonarlint-visualstudio#878]
### Description

I'm getting a false positive on S1854

### Repro steps

```csharp
    public class Foo
    {
        public void Bar()
        {
            bool isFirst = true;
            foreach (var i in Enumerable.Range(1, 10))
            {
                try
                {
                    if (isFirst)
                    {
                        Console.WriteLine("First");
                    }
                    else
                    {
                        Console.WriteLine("Not first");
                    }
                }
                finally
                {
                    isFirst = false;
                }
            }
        }
    }
```

### Expected behavior

No warning

### Actual behavior

![image](https://user-images.githubusercontent.com/130373/56343966-8be07600-61bc-11e9-9f92-ebe477a37ad4.png)

### Known workarounds

n/a

### Related information

* SonarLint for VS Version: 4.9.0.4316
* Visual Studio 2019 / C#

**Comment by remcoros** [comment]
Created the issue in the right repo now: https://github.com/SonarSource/sonar-dotnet/issues/2393
